assets/js/modules/analytics-4/datastore/settings.js

```diff
@@ -77,24 +83,41 @@ export async function submitChanges( { select, dispatch } ) {
 	}
 
 	let webDataStreamID = select( MODULES_ANALYTICS_4 ).getWebDataStreamID();
-	if (
-		propertyID &&
-		( webDataStreamID === WEBDATASTREAM_CREATE ||
-			! isValidWebDataStreamID( webDataStreamID ) )
```

Comment:
Now that we require a web data stream name for the creation of a web data stream, I think it no longer makes sense to autonomously attempt the creation of a web data stream if it is invalid. The behaviour should rather be that the submission of the form should be prevented if such an invalid web data stream is selected.

Reply:
SGTM and makes sense. Nice!